### PR TITLE
add dataset training & masked metric logging

### PR DESCRIPTION
**Pull request by msindeeva**
#### What does this implement/fix? Explain your changes.
Adds dataset config parsing and masked metric computation

#### What testing did you do to verify the changes in this PR?
Ran training script with old & new configs

### CODE REVIEW

File: selene_sdk/utils/config_utils.py [hunk 1]
```diff
@@ -132,6 +137,124 @@ def initialize_model(model_configs, train=True, lr=None):
     return model, criterion
 
 
+def create_data_source(configs, output_dir=None, load_train_val=True, load_test=True):
+    """
+    Construct data source(s) used in training/evaluation.
```
Comment (avetisianms) [inline selene_sdk/utils/config_utils.py:142]: can you please elaborate, as this comment doesn't really describe what the method is doing

File: selene_sdk/utils/config_utils.py [hunk 2]
```diff
@@ -132,6 +137,124 @@ def initialize_model(model_configs, train=True, lr=None):
...
+        sampler = instantiate(sampler_info)
+        return sampler
+    if "dataset" in configs:
+        dataset_info = configs["dataset"]
+        train_intervals = []
+        val_intervals = []
+        test_intervals = []
+        with open(dataset_info["sampling_intervals_path"]) as f:
+            for line in f:
+                chrom, start, end = line.rstrip().split("\t")[:3]
```
Comment (avetisianms) [inline selene_sdk/utils/config_utils.py:181]: what if there are more than 3 fields in the line?

**Review by avetisianms — APPROVED** [review]
good code, please go forward and commit it after addressing minor comments that I had

File: selene_sdk/utils/config_utils.py (hunk 2 above)
Comment (msindeeva) [reply selene_sdk/utils/config_utils.py:181]: 1. It doesn't matter, I'm only interested in interval regions specified in the file, not any additional info. And `[:3]` should take care of that anyway
2. By design of this file (generated by [this script](https://github.com/sberbank-ai-lab/DeepCT/blob/master/data/generate_target_files.py)) it shouldn't contain anything but intervals

File: selene_sdk/utils/config_utils.py (hunk 1 above)
Comment (msindeeva) [reply selene_sdk/utils/config_utils.py]: fixed in 0074139

**Pull request merged**